### PR TITLE
Tweak: Controls System Optimization (ED-3378, ED-3643, ED-3744, ED-3814, ED-3839, ED-3855)

### PR DESCRIPTION
**Pull request by udidol**

### CODE REVIEW

File: core/breakpoints/manager.php [hunk 1]
```diff
@@ -20,9 +20,50 @@ class Manager extends Module {
...
+	 * -- 'on': Responsive controls are duplicated in `add_responsive_control()`.
+	 * -- 'off': Responsive controls are NOT duplicated in `add_responsive_control()`.
+	 * -- 'dynamic': Responsive controls are only duplicated if their config contains `'dynamic' => 'active' => true`.
+	 *
+	 * When generating Post CSS, the mode is set to 'on'. When generating Dynamic CSS, the mode is set to 'dynamic'.
+	 *
+	 * default value is 'off'.
+	 *
+	 * @since 3.4.0
+	 * @access public
```
Comment (kobizz) [inline core/breakpoints/manager.php:61]:
```suggestion
	 * @access private
```

File: core/breakpoints/manager.php [hunk 2]
```diff
@@ -183,6 +224,51 @@ public static function get_default_config() {
 		];
 	}
 
+	public function get_breakpoints_config() {
```
Comment (kobizz) [inline core/breakpoints/manager.php:227]: Add doc

File: core/experiments/manager.php [hunk 3]
```diff
@@ -268,6 +268,18 @@ private function add_default_features() {
 			'release_status' => self::RELEASE_STATUS_BETA,
 			'default' => self::STATE_ACTIVE,
 		] );
+
+		$this->add_feature( [
+			'name' => 'additional_custom_breakpoints',
+			'title' => __( 'Additional Custom Breakpoints', 'elementor' ),
+			'description' => __( 'Add Additional Custom Breakpoints (beyond just \'mobile\' and \'tablet\') Optimize your site\'s performance when using responsive controls.', 'elementor' )
+							. '<br /><strong>' . __( 'Please note! Conditioning controls on values of responsive controls is not supported when this mode is active.', 'elementor' ) . '</strong>',
```
Comment (kobizz) [inline core/experiments/manager.php:276]: Use `esc_html__` for translations

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/experiments/manager.php (hunk 3 above)
Comment (udidol) [reply core/experiments/manager.php:276]: Done.

File: core/breakpoints/manager.php (hunk 2 above)
Comment (udidol) [reply core/breakpoints/manager.php:254]: Done.

File: core/breakpoints/manager.php (hunk 1 above)
Comment (udidol) [reply core/breakpoints/manager.php]: Done.

File: includes/editor-templates/panel.php [hunk 4]
```diff
@@ -225,14 +225,25 @@
...
+			} else {
+				devicesForDisplay.unshift( 'desktop' );
+			}
+
+			var devices = responsive.devices || devicesForDisplay;
 
 			_.each( devices, function( device ) {
-				var deviceLabel = device.charAt(0).toUpperCase() + device.slice(1),
+				// The 'Desktop' label is made accessible via the global config because it needs to be translated.
+				var deviceLabel = 'desktop' === device ? elementor.config.responsive.desktop_label : activeBreakpoints[ device ].label,
```
Comment (kobizz) [inline includes/editor-templates/panel.php:242]:
```suggestion
				var deviceLabel = 'desktop' === device ? '<?php echo __( 'Desktop', 'elementor' ); ?>': activeBreakpoints[ device ].label,
```

File: core/editor/editor.php [hunk 5]
```diff
@@ -588,6 +588,12 @@ public function enqueue_scripts() {
 			],
 			// Empty array for BC to avoid errors.
 			'i18n' => [],
+			// 'responsive' contains the custom breakpoints config introduced in Elementor v3.2.0
+			'responsive' => [
+				'breakpoints' => Plugin::$instance->breakpoints->get_breakpoints_config(),
+				'icons_map' => Plugin::$instance->breakpoints->get_responsive_icons_classes_map(),
+				'desktop_label' => __( 'Desktop', 'elementor' ),
```
Comment (kobizz) [inline core/editor/editor.php:595]: You don't need this variable. See below

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/editor/editor.php (hunk 5 above)
Comment (udidol) [reply core/editor/editor.php]: Done.

File: includes/editor-templates/panel.php (hunk 4 above)
Comment (udidol) [reply includes/editor-templates/panel.php]: Done.

File: assets/dev/scss/editor/_responsive-bar.scss [hunk 6]
```diff
@@ -138,3 +138,6 @@
 	}
 }
 
+i.eicon-tilted {
```
[the remark this answers is not in the document]
Comment (udidol) [reply assets/dev/scss/editor/_responsive-bar.scss]: Done.

File: assets/dev/scss/direction/editor-dark-mode.scss [hunk 7]
```diff
@@ -455,7 +455,7 @@ a.elementor-responsive-switcher {
 	}
 }
 
-$devices: 'desktop', 'tablet', 'mobile';
+$devices: 'widescreen', 'desktop', 'laptop', 'tablet_extra', 'tablet', 'mobile_extra', 'mobile';
```
[the remark this answers is not in the document]
Comment (udidol) [reply assets/dev/scss/direction/editor-dark-mode.scss]: Done.

**Pull request merged**